### PR TITLE
Update OpenSSL to v1.1.1b

### PR DESCRIPTION
**Pull request by vsebe**
Add to PATH via profile for remote desktop connections and via ssh client environment for Jenkins agent.
Link libraries and clean up old versions.

Signed-off-by: Violeta Sebe <vsebe@ca.ibm.com>

### CODE REVIEW

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/OpenSSL/tasks/main.yml [hunk 1]
```diff
@@ -1,64 +1,184 @@
...
+    - openssl_version.stdout == ''
+  ignore_errors: True
+  tags: openssl
+
+- name: Create /etc/ld.so.conf.d/openssl-{{ openssl_latest_release }}.conf
+  copy:
+    dest: /etc/ld.so.conf.d/openssl-{{ openssl_latest_release }}.conf
+    content: /usr/local/openssl-{{ openssl_latest_release }}/lib
+  when:
+    - (ansible_distribution == "RedHat" or ansible_distribution == "CentOS" or ansible_distribution == "Ubuntu" or ansible_distribution == "SLES")
```
Comment (jdekonin) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/OpenSSL/tasks/main.yml]: Can you specify which os / release variants you tested this against?

Comment (vsebe) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/OpenSSL/tasks/main.yml]: centos 6, rhel6, ubuntu14, ubuntu16, sles12

**Comment by sxa555** [comment]
Hi, what is the driver for this change (i.e. is there a related issue) and are there any other related PRs to handle the fact this directory name will now be different from what it was before?

**Comment by vsebe** [comment]
@jdekonin ready for review, thank you!

**Comment by sxa555** [comment]
@vsebe See my question from yesterday :-)

**Comment by vsebe** [comment]
@sxa555 - the directory name change has no impact on OpenJ9 build and tests.
This PR also sets openssl-1.1.1 as default for the Jenkins user (see the tasks that add openssl to the PATH) on Linux x86 platforms.

**Comment by sxa555** [comment]
I'm really not comfortable with changing the default like that - why is that needed instead of just doing it in whatever jobs require it? That has the potential to lead to confusion.

**Comment by vsebe** [comment]
@sxa555 - to add it to the jobs would have update either multiple Jenkins files across multiple repositories or update the slaves configuration.
Setting it as default could be made optional.

**Comment by sxa555** [comment]
@vsebe Talking to @mpirvu in slack he says that the JITaaS code will build against 1.0.2 which is available on most machines anyway, so I'm even less clear about why we need 1.1.1 at all just now. Can you clarify please?

**Comment by jdekonin** [comment]
JITaaS code will build against v1.0.2 or v1.1.1 long term releases, but since v1.0.2 is end of life at the end of 2019 as per [openssl.org](https://www.openssl.org/policies/releasestrat.html), it is likely that OpenJ9 will be using v1.1.1 as the requirement.  @pshipton or @mpirvu, please state otherwise if that isn't the case.

Could we not have this role included but maybe not added to main.yml until such time that it is something that AdoptOpenJDK farm machines would require?  I would suggest that this role get a different tag though, say `openssl-1.1.1`?

**Comment by sxa555** [comment]
> Could we not have this role included but maybe not added to main.yml 

Yeah I would be ok with that :-)

Ref the support statement, that doesn't prevent the distributions from continuing to support it beyond that. I haven't looked to see if RedHat have made a statement on it yet but I doubt RHEL7 will be upgraded beyond 1.0.2 even if official support at openssl does not continue, so it will still be the default supported version on many systems, including our build ones.

**Comment by mpirvu** [comment]
As stated in a different item my preference would be to build JITaaS code against openssl 1.1.x because:
(1) That's what we filed a CQ for
(2) We have done much more testing with 1.1.x than with 1.0.x
(3) At some point 1.0.x will become obsolete.

**Comment by vsebe** [comment]
@jdekonin @sxa555 ready for your review, thank you!

**Comment by sxa555** [comment]
@mpirvu Just for clarification when you refer to "CQ" is that the [Eclipse Contribution Questionnaire](https://wiki.eclipse.org/Development_Resources/Contribution_Questionnaire)?

**Comment by mpirvu** [comment]
> @mpirvu Just for clarification when you refer to "CQ" is that the Eclipse Contribution Questionnaire?

Yes.

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/OpenSSL/tasks/main.yml [hunk 2]
```diff
@@ -1,64 +1,252 @@
...
+    owner: "{{ Jenkins_Username }}"
+    mode: 0644
+  when:
+    - (ansible_distribution == "RedHat" or ansible_distribution == "CentOS" or ansible_distribution == "Ubuntu" or ansible_distribution == "SLES")
+    - ansible_architecture == "x86_64"
+    - openssl_version.stdout == ''
+  tags: openssl-1.1.1
+
+- name: Add OpenSSL-1.1.1b to Jenkins user's environement
+  shell: echo PATH=/usr/local/openssl-1.1.1b/bin:$PATH > /home/{{ Jenkins_Username }}/.ssh/environment
```
Comment (AdamBrousseau) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/OpenSSL/tasks/main.yml]: Should `openssl-1.1.1b` be hardcoded here?

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/OpenSSL/tasks/main.yml [hunk 3]
```diff
@@ -1,64 +1,252 @@
...
+    - openssl_version.stdout == ''
+  tags:
+    - openssl
+    - skip_ansible_lint
+
+- name: Enable user environment in the OpenSSH daemon configuration file
+  replace:
+    path: /etc/ssh/sshd_config
+    regexp: '^PermitUserEnvironment no'
+    replace: 'PermitUserEnvironment yes'
```
Comment (AdamBrousseau) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/OpenSSL/tasks/main.yml]: If the file doesn't have this line, will it be added?

**Pull request merged**